applications/luci-app-wireguard/luasrc/view/wireguard.htm

```diff
@@ -4,135 +4,41 @@
 -%>
 
 <%
-  function strip(s)
-    return (s:gsub("^%s*(.-)%s*$", "%1"))
-  end
-
-  function is_valid(s)
-    return (string.len(strip(s)) > 0)
-  end
-
-  function peer_add(peers, public_key)
-    table.insert(peers, {
-      public_key = public_key,
-      endpoint = "",
-      transfer_rx = 0,
-      transfer_tx = 0,
-      latest_handshake = -1,
-      persistent_keepalive = "",
-      allowed_ips = { }
-    })
-  end
-
-  function peer_set_transfer(peers, public_key, rx, tx)
-    for key, peer in pairs(peers) do
-      if peer.public_key == public_key then
-        peers[key].transfer_rx = rx
-        peers[key].transfer_tx = tx
-        break
-      end
-    end
-  end
-
-  function peer_set_latest_handshake(peers, public_key, latest_handshake)
-    for key, peer in pairs(peers) do
-      if peer.public_key == public_key then
-        peers[key].latest_handshake = latest_handshake
-        break
-      end
-    end
-  end
-
-  function peer_set_endpoint(peers, public_key, endpoint)
-    for key, peer in pairs(peers) do
-      if peer.public_key == public_key then
-        peers[key].endpoint = endpoint
-        break
-      end
-    end
-  end
-
-  function peer_set_persistent_keepalive(peers, public_key, persistent_keepalive)
-    for key, peer in pairs(peers) do
-      if peer.public_key == public_key then
-        peers[key].persistent_keepalive = persistent_keepalive
-        break
-      end
-    end
-  end
-
-  function peer_set_allowed_ips(peers, public_key, allowed_ips)
-    for key, peer in pairs(peers) do
-      if peer.public_key == public_key then
-        for ipkey, ipvalue in pairs(string.split(allowed_ips, " ")) do
-          if is_valid(ipvalue) then
-            table.insert(peers[key].allowed_ips, strip(ipvalue))
-          end
+  local data = { }
+  local last_device = ""
+
+  local wg_dump = luci.sys.exec("wg show all dump")
+  for key, line in pairs(string.split(wg_dump, "\n")) do
```

Comment:
consider using `io.popen()` and `lines()` here, this avoids the (very expensive) splitting operation.

    local wg_dump = io.popen("wg show all dump")
    if wg_dump then
        local line
        for line in wg_dump:lines() do
            -- ...
        end
    end